packages/shared/src/hooks/squads/usePublicSquadRequests.ts

```diff
@@ -31,8 +71,31 @@ export const usePublicSquadRequests = (): UsePublicSquadRequestsResult => {
       },
     });
 
+  const edges = requests?.pages[requests?.pages?.length - 1]?.edges;
+  const latestRequest = edges?.[edges?.length - 1]?.node;
+
+  const status = useMemo(() => {
+    if (!latestRequest) {
+      return SquadStatus.InProgress;
+    }
+
+    if (latestRequest.status === PublicSquadRequestStatus.Rejected) {
+      const fourteenDaysAgo = subDays(new Date(), 14);
+      const requestDate = new Date(latestRequest.createdAt);
+      return fourteenDaysAgo < requestDate
+        ? SquadStatus.Rejected
+        : SquadStatus.InProgress;
+    }
```

Comment:
Not sure what was discussed on RFC, but why do we do this logic on the frontend, feels like this should be handled on API side?
CC: @capJavert

Reply:
This is not going to be used for validation but something we need to display some alert.